src/client/api/rpc.c

```diff
@@ -122,6 +123,16 @@ query_cb(struct crt_proto_query_cb_info *cb_info)
 		nr_ranks = dc_mgmt_net_get_num_srv_ranks();
 		D_ASSERT(nr_ranks > 0);
 		rproto->rank_idx = (rproto->rank_idx + 1) % nr_ranks;
+
+		/** We made a full loop and found no engines alive */
+		if (rproto->start_rank_idx == rproto->rank_idx) {
+			D_ERROR("crt_proto_query_with_ctx() failed -- All %d targets tried\n",
+				nr_ranks);
+			rproto->rc        = cb_info->pq_rc;
```

Comment:
Not sure to understand this assignment.
Is it going to inform the caller that the query failed ?